apps/heft/src/plugins/TypeScriptPlugin/Tslint.ts

```diff
@@ -0,0 +1,152 @@
+import { Terminal, JsonFile, Colors } from '@rushstack/node-core-library';
+import { IExtendedFileSystem } from '../../utilities/fileSystem/IExtendedFileSystem';
+import { ResolveUtilities } from '../../utilities/ResolveUtilities';
+import { IExtendedLinter } from './internalTypings/TslintInternals';
+
+interface ITslintOptions extends ILinterBaseOptions {
+  tslintPackagePath: string;
+
+  fileSystem: IExtendedFileSystem;
+}
+
+export class Tslint extends LinterBase<TTslint.RuleFailure> {
+  private readonly _tslint: typeof TTslint;
+  private readonly _fileSystem: IExtendedFileSystem;
+
+  private _tslintConfiguration: TTslint.Configuration.IConfigurationFile;
+  private _linter: IExtendedLinter;
+  private _enabledRules: TTslint.IRule[];
+  private _ruleSeverityMap: Map<string, TTslint.RuleSeverity>;
+  protected _lintResult: TTslint.LintResult;
+
+  public constructor(options: ITslintOptions) {
+    super('tslint', options);
+
+    this._tslint = require(options.tslintPackagePath);
+    this._fileSystem = options.fileSystem;
+  }
+
+  public static getConfigHash(
+    configFilePath: string,
+    terminal: Terminal,
+    fileSystem: IExtendedFileSystem
+  ): crypto.Hash {
+    interface IMinimalConfig {
+      extends?: string;
+    }
+
+    terminal.writeVerboseLine(`Examining config file "${configFilePath}"`);
+
+    const rawConfig: string = fileSystem.readFile(configFilePath);
+    const parsedConfig: IMinimalConfig = JsonFile.parseString(rawConfig);
+    let hash: crypto.Hash;
+    if (parsedConfig.extends) {
+      const extendsFullPath: string = ResolveUtilities.resolvePackagePath(
+        parsedConfig.extends,
+        path.dirname(configFilePath)
+      );
+      hash = Tslint.getConfigHash(extendsFullPath, terminal, fileSystem);
+    } else {
+      hash = crypto.createHash('sha1').update(rawConfig);
+    }
+
+    return hash.update(rawConfig);
+  }
+
+  public printVersionHeader(): void {
+    this._terminal.writeLine(`Using TSLint version ${this._tslint.Linter.VERSION}`);
+  }
+
+  public reportFailures(): void {
+    if (this._lintResult.failures?.length) {
+      this._terminal.writeWarningLine(
+        `Encountered ${this._lintResult.failures.length} TSLint error${
+          this._lintResult.failures.length > 1 ? 's' : ''
+        }:`
+      );
+      for (const tslintFailure of this._lintResult.failures) {
+        const buildFolderRelativeFilename: string = path.relative(
+          this._buildFolderPath,
+          tslintFailure.getFileName()
+        );
+        const { line, character } = tslintFailure.getStartPosition().getLineAndCharacter();
+        const severity: string = tslintFailure.getRuleSeverity().toUpperCase();
+        this._terminal.writeWarningLine(
+          '  ',
+          Colors.yellow(`${severity}: ${buildFolderRelativeFilename}:${line + 1}:${character + 1}`),
+          ' - ',
+          Colors.yellow(`(${tslintFailure.getRuleName()}) ${tslintFailure.getFailure()}`)
```

Comment:
FYI ESLint's `--format unix` output uses a `:` instead of ` - `:
```
src\index.ts:36:18: Interface name `MissingI` must match the RegExp: /^_?I[A-Z]/ [Error/@typescript-eslint/naming-convention]
```

I think I might prefer your hyphen, if the various editors recognize it.  I definitely do not like ESLint's way of using `Error/` to indicate an error.  Our `ERROR: ` prefix looks better.

Reply:
Yeah I didn't like ESLint's default formatting. VSCode seems fine with it, although I'm not sure if we want to test any specific editor detection?

Reply:
The other popular ones where I work are: Sublime, WebStorm, Atom

VIM and IntelliJ Ultimate are other possibilities.

Seems time consuming to install them all and try them, though...